File: PlatformAllocation/PlatformAllocation.java
import java.util.*;

class PlatformAllocation{

	public static void main(String[] args) {


        Scanner sc =new Scanner(System.in);
		int n= sc.nextInt();
		Vector<String> trainName = new Vector<String>(), arrivalTime = new Vector<String>(), departureTime = new Vector<String>(), timeTakenToDepart = new Vector<String>();
		Vector<Integer> trainNumber = new Vector<Integer>();
		for(int i=0;i<n;i++)
		{
			trainName.add(sc.next());
			trainNumber.add(sc.nextInt());
			arrivalTime.add(sc.next());
			departureTime.add(sc.next());
			timeTakenToDepart.add(sc.next());
		}
		Solution so = new Solution();
		Pair p=so.platformAllocation(trainName,trainNumber,arrivalTime,departureTime,timeTakenToDepart);
		System.out.println(p.first);
		for(int i=0;i<n;i++)
		{
			System.out.println(trainName.get(i)+" "+trainNumber.get(i)+" "+(p.second).get(i));
		}
	}
}

File: PlatformAllocation/PlatformAllocation.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"


int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    int n;
    cin >> n;
    vector<string> trainName(n), timeOfArrival(n), timeOfDeparture(n), timeToDepart(n);
    vector<int> trainNum(n);

    for(int i = 0 ; i < n ; ++i) {
        cin >> trainName[i] >> trainNum[i] >> timeOfArrival[i] >> timeOfDeparture[i] >> timeToDepart[i];

    }

    pair<int, vector<int> > result = assignPlatformToTrain(trainName, timeOfArrival, timeOfDeparture, timeToDepart, trainNum);

    int platformsRequired = result.first;
    vector<int> platformAssigned = result.second;

    cout << platformsRequired << "\n";
    for(int i = 0 ; i < n ; ++i) {
        cout << trainName[i] << " " << trainNum[i] << " " << platformAssigned[i] << "\n";
    }
}
